inc/namespace.php

```diff
@@ -135,13 +135,48 @@ function get_json_file_contents_as_array( $file ) : array {
 
 	if ( json_last_error() !== JSON_ERROR_NONE ) {
 		// phpcs:ignore
-		trigger_error( json_last_error_msg(), E_USER_WARNING );
+		trigger_error( 'composer.json could not be parsed\n' . json_last_error_msg(), E_USER_WARNING );
 		return [];
 	}
 
+	$modules = Module::get_all();
+
+	// Unregistered module with no "entrypoint" property.
+	foreach ( $contents['extra']['altis']['modules'] as $key => $module ) {
+		if ( ! array_key_exists( $key, $modules ) && ! array_key_exists( 'entrypoint', $module ) ) {
+			log_error( 'Please specify an entry point setting for your custom module.' );
+		}
+	}
+
+	// Registered module config not under "modules" property.
+	foreach ( $contents['extra']['altis'] as $key => $config ) {
+		if ( array_key_exists( $key, $modules ) ) {
+			log_error( 'Please specify an entry point setting for your custom module.' );
+		}
+
+		//Custom module at top level with "entrypoint" property:
+		if ( array_key_exists( 'entrypoint', $config ) ) {
+			log_error( 'Module configurations should be defined under altis.extra.modules' );
+		}
+	}
+
 	return $contents;
 }
 
+/**
+ * Throws exception or triggers error depending on environment.
+ *
+ * @param string $msg
+ * @param int $level
+ */
+function log_error( string $msg, int $level = E_USER_ERROR ) {
```

Comment:
`E_USER_ERROR` will cause PHP to stop executing, hence why we use `E_USER_WARNING` generally